invoker/core/src/main/java/com/google/cloud/functions/invoker/CloudFunctionsContext.java

```diff
@@ -46,6 +48,9 @@
   @Nullable
   public abstract String resource();
 
+  // TODO: expose this in the Context interface (as a default method).
+  abstract Map<String, String> params();
```

Comment:
We're going to need _some_ way to get at the `params` field from the legacy JSON. This is actually a hole in the current GCF Java API. I don't think users have any way to get at the `params` from within a function, since they aren't reflected in the `Context` object that we pass. Hence the TODO, because I think we should plug that hole.

Reply:
But if it's not going to be part of the CloudEvent, I don't think it's a good idea to plug the hole in the legacy event format either. If users can't get at it now, they can't miss it when it's not in a CloudEvent. The expectation from the Firebase team is that this is best handled by library code which parses the resource name.